clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/WebSocketTransport.java

```diff
@@ -13,35 +16,47 @@
     private OnReceiveCallBack onReceiveCallBack;
     private URI url;
     private Logger logger;
+    private String accessToken;
 
     private static final String HTTP = "http";
     private static final String HTTPS = "https";
     private static final String WS = "ws";
     private static final String WSS = "wss";
+    private static final String ACCESS_TOKEN_PREFIX = "access_token=";
 
     public WebSocketTransport(String url, Logger logger) throws URISyntaxException {
         this.url = formatUrl(url);
         this.logger = logger;
     }
 
-    public URI getUrl(){
+    public URI getUrl() {
         return url;
     }
 
     private URI formatUrl(String url) throws URISyntaxException {
-        if(url.startsWith(HTTPS)){
+        if (url.startsWith(HTTPS)) {
             url = WSS + url.substring(HTTPS.length());
-        }
-        else if(url.startsWith(HTTP)){
+        } else if (url.startsWith(HTTP)) {
             url = WS + url.substring(HTTP.length());
         }
+        if (url.indexOf(ACCESS_TOKEN_PREFIX) > 0) {
```

Comment:
Why does this exist?

Reply:
We parse the access token from the url so we can put it in a header when we construct the underlying WebSocketClient. We actually know the accessToken before we even contruct the transport. I can add a ctor that takes headers and get rid of this

Reply:
That would be much cleaner.